### DISCUSSION TITLE
Central repo vs private block-chain vs public block-chain

### DISCUSSION THREAD

**Discussion by fretchen** [openkfw/TruBudget#1400] [General]
From my understanding TruBudget is running on a private blockchain solution called [Multi-Chain](https://www.multichain.com/). This brings up a number of questions for me if I compare it to other approaches to store data, namely classical repos and public block-chains.

When I try to evaluate them I have a hard time to judge. So I would be very interested in the feedback.

## Central repo

The most classic approach to store data would be quite often the storage of data in some cloud storage like Dropbox, nextcloud or sharepoint. Right ? Looking into this option I would see the following things:

- Easy to set up quite often.
- No need to think about it anymore.
- All I need is a trusted/neutral third-party.

When is this approach clearly failing compared to a private block-chain ?

## Public block-chain

The progress of Euthereum etc makes you wonder if a public block-chain would not be interesting in certain use-cases. What makes it tempting to think about this is:

- enormous number of devs over there.
- no need to set up nodes at all
- no trust needed at all.

However, is there any example where a public block-chain has been used a decentralized workflow tool ? What would be needed there and which part of TruBudget would be hard to transfer onto the public block-chain right now ?

## Private block-chain

- How is the question of trust handled in multichain private block-chain ? In other words, how much trust do I have to have into each node operator (that the node operator is not altering data or simply trying to steal data). 

## Mitigation methods

How would I minimize the risk of betting on the "wrong" stack  in some specific usecases ? In other words how interchangeable are these three alternatives of storage in the end ?

**Comment by jzakotnik** [comment]
Yeah, definitely a good structure. My take would be independent from the product multichain:

## Central repository
Central repository is always an option. In a network of N participants, I would separate it into the question if one of the participants can be the data provider or of a 3rd party intermediary is needed. The technology implementation is from my view less relevant than the governance - who owns the dropbox, onedrive, you-name-it account, who has admin rights there, who has access to the data?
Advantage is the simplicity of the handling. Disadvantage is the someone has more access than others and the risk to manipulate the data (on purpose or e.g. via cyberattacks). 

## Decentralised repository

Here we have some options, for example:
- Decentralised database: We analysed various database products that offer decentralisation, but it is always implemented for scalability and performance (so it is rather distribution than decentralisation). In the end the same applies as to a centralised solution.
- Private, permissioned DLT: The network decides together which level of trust they require and implement the relevant parameters (e.g. consensus mechanism, node permissions, etc). The question is not how much trust I have into the node operator, but from my view, you decide the level of trust a-priori and the configure the DLT accordingly. Advantage is that the network of participants can decide together how to handle data. The disadvantage is that there's significant effort to build your own DLT, maintain it etc.
- Open, permissionless DLT, e.g. ethereum, polygon. This is using a data layer like a SaaS-solution, which means you take the configuration parameters and the network as is and you are only able to configure some parts of it, e.g. your specific smart contract. Advantage is minimal configuration and mature ecosystem of tools, people and support. Disadvantage are the fees and that all data is visible to everyone. My personal take is, that only the open systems will survive and you can see in the emissions of crypto-papers (see [Siemens](https://press.siemens.com/global/de/pressemitteilung/siemens-begibt-erstes-elektronisches-wertpapier-auf-der-blockchain) ) that this is realistic) and there's a huge advantage in open systems. The disadvantage is that it is difficult to store sensitive data on open DLT.

## Hybrid repository (centralised/decentralised)
Many DLT implementations use centralised or private DLT and use a public layer only to store hashes etc. This only secures that data is not manipulated, but is usually difficult to use and e.g. if the data is deleted, you can detect this but you cannot do anything against it. Nevertheless I believe that this is a viable solution, to store as much data as possible on a public blockchain and revert to private or centralised systems only for definitely sensitive data (e.g. via links). To my knowledge there is no public blockchain workflow engine available, but if OpenData is took seriously, a majority of information could be saved on open DLT.
